QuizApp/Presenter/QuizzesViewPresenter.swift

```diff
@@ -0,0 +1,107 @@
+//
+//  QuizzesViewPresenter.swift
+//  QuizApp
+//
+//  Created by Antonio Markotic on 16.05.2021..
+//
+
+
+import UIKit
+import CoreData
+
+
+class QuizzesViewPresenter{
+    private let nService : NetworkService
+    weak private var quizzesViewDelegate : QuizzesViewController?
+    
+    let repository = QuizDataRepository(networkDataSource: NetworkDataSource(), coreDataSource: QuizCoreDataSource(coreDataContext: CoreDataStack(modelName: "QuizAppModel").managedContext))
+    //array kvizova
+    var quizzArray :[Quiz] = []
+    
+    //array kategorija
+    var uniqueSectionArray = [QuizCategory]()
+    
+    //2-d matrica popunjena s kvizovima sortiranim prema kategorijama
+    var matrix = [[Quiz]]()
+    var colorOfCategory =  [UIColor]()
+    
+    var firstLoad = false
+    
+    init(networking: NetworkService){
+        self.nService = networking
+    }
+    func setquizzesViewDelegate(quizzesViewDelegate : QuizzesViewController?){
+        self.quizzesViewDelegate = quizzesViewDelegate
+    }
+    
+    
+    
+    
+    func fetchQuizzes(with internetConnection: Bool){
```

Comment:
This logic of fetching data from API and saving it to CoreData should be in Repository. The presentation layer should not be aware of this data handling - that is the responsibility of the Data layer.